extensions-core/hdfs-storage/src/main/java/org/apache/druid/inputsource/hdfs/HdfsInputSource.java

```diff
@@ -101,20 +102,25 @@ public HdfsInputSource(
     return paths;
   }
 
-  public static Collection<Path> getPaths(List<String> inputPaths, Configuration configuration) throws IOException
+  public static Collection<Path> getPaths(List<String> inputPathStrings, Configuration configuration) throws IOException
   {
-    if (inputPaths.isEmpty()) {
+    if (inputPathStrings.isEmpty()) {
       return Collections.emptySet();
     }
 
     // Use FileInputFormat to read splits. To do this, we need to make a fake Job.
     Job job = Job.getInstance(configuration);
 
     // Add paths to the fake JobContext.
-    for (String inputPath : inputPaths) {
+    for (String inputPath : inputPathStrings) {
       FileInputFormat.addInputPaths(job, inputPath);
     }
 
+    final Path[] inputPaths = FileInputFormat.getInputPaths(job);
+    if (Arrays.stream(inputPaths).anyMatch(path -> !"hdfs".equalsIgnoreCase(path.toUri().getScheme()))) {
+      throw new IllegalArgumentException("Input paths must be the HDFS path");
+    }
+
```

Comment:
It appears that `FileInputFormat#addInputPath` already has it's own scheme validation, so I don't believe we need this check. See `org.apache.hadoop.fs.FileSystem#checkPath`

Reply:
`FileSystem` has a couple of implementations such as `DistributedFileSystem`, `S3AFileSystem`, etc which support different schemes. The default file system is `LocalFileSystem` unless you set `fs.default.name`. I think we should not allow anything else but only `hdfs` and let users use the right inputSource to read from other storage (e.g., `S3InputSource to read from s3).

Reply:
Are there users that rely on this behavior? Would restricting this to just hdfs mean it's not possible for some users to ingest files from these locations any more? If that's the case, should we introduce a config to allow server admins to specify which schemes are supported?

Reply:
@jihoonson Technically `WebHdfsFileSystem` did work with this inputsource before, and so it could break ingestion pipelines for operators relying on HDFS Inputsource with `webhdfs` scheme.  Could you please comment on what is the motivation behind restricting to only `hdfs`?

Reply:
@suneet-s @a2l007 good point. I realized that this is a [documented feature](https://druid.apache.org/docs/0.20.0/ingestion/native-batch.html#hdfs-input-source) that should be supported. The concern here is that users can use whatever protocol they want if it's supported by the hdfs client. Instead of restricting it to only hdfs, I added new configs for http and hdfs inputSources so that users can configure what protocols they want to allow.